p2p/3.IPFS_or_HTTP/README.md

```diff
@@ -0,0 +1,149 @@
+>This is how HTTP works. When retrieving data, HTTP focuses on **location**. Have you noticed the url ?
+First, it is 0.0.0.0 which is your local IP; then :8000 to signify the port. Finally, the routes or file you want to
+get, joined by a /. That format does not ring a bell to you ? It is like a path !
+
+## Step 2 - IPFS
+
+>HTTP is cool but has its limits : if the server is down, you won't be able to retrieve the data stored. Furthermore,
+your government can easily block access to certain servers by their IPs that host particular website for censure purposes.
+Let's see how IPFS answers this issues.
+
+At its core, IPFS is a **distributed** system for storing and accessing files, websites, applications, and data.
+Instead of referring to data (photos, articles, videos) by **location**, or which server they are stored on, IPFS refers to everything by that data’s hash, meaning the **content itself.**
+
+The idea is that if you want to access a particular page from your browser, IPFS will ask the entire network, “does anyone have the data that corresponds to this hash?” A node on IPFS that contains the corresponding hash will return the data, allowing you to access it from anywhere (and potentially even offline).
+
+If this is not enough clear for you, I strongly advise you to refer to [this](https://www.youtube.com/watch?v=5Uj6uR3fp-U) video.
+
+### 2.0 Improve the storage
+
+Here is what we are going to do : We are going to upload our files directly on IPFS and not locally anymore.
+Instead of having the file locally, let's have its corresponding hash in our database.
+
+1. Go to `musicshare/models.py` and add a CharField for the hash.
+2. Go to `musicshare/views.py` and modify the code of the upload view to communicate with IPFS API in order to upload the file there.
+   Please use port `5001` for the connection.
+3. Open `musicshare/templates/musicshare/index.html` to line `179` and make sure the hash of the song appears.
+
+To apply your migrations, shut down the server and relaunch it.
+
+Test another time to upload your music, copy past the hash of the newly added song and go to https://ipfs.infura.io:5001/api/v0/cat?arg={your_hash_here}.
+You should see your song play, even if your server has been shut down !
+
+### 2.2 Retrieve
+Last step : if anyone wants to download from our website some mp3 songs, we need to get it from IPFS.
+Since you did the previous step, this one would seem easy : in your `musicshare/view.py` on the `download` view, 
+do the same thing as previously but instead of adding a file, call the `cat` method.
+
+## Going further
+A very cool feature with IPFS is that if someone is having an IPFS node running on its machine and download your mp3 audio
+then you deleted it, you will be able to retrieve it from its node !
+
+[Here](https://docs.ipfs.io/how-to/command-line-quick-start) is how you can deploy and configure your own IPFS node.
```

Comment:
Where are authors and bonuses and organization? Check that [workshop](https://github.com/PoCInnovation/Workshops/tree/master/software/17.gRPC#rocket-to-go-further) for example